### PR TITLE
Upgrade template

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Releasing is hard

### How was it fixed?

Fixes the release script by merging the template updates.

Updates include:
* Release script has been updated to check the git remote properly
* better words in docs, comments

### Todo:
- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-utils/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by reedsa** [comment]
I noticed we dont have linting configured for Python 3.7 out but still build the wheel. Is that intentionally skipped?

Also see the versions for flake8, etc are outdated but that could be another issue.

File: README.md [hunk 1]
```diff
@@ -55,7 +54,7 @@ The version format for this repo is `{major}.{minor}.{patch}` for stable, and
 
 To issue the next version in line, specify which part to bump,
 like `make release bump=minor` or `make release bump=devnum`. This is typically done from the
-master branch, except when releasing a beta (in which case the beta is released from master,
+main branch, except when releasing a beta (in which case the beta is released from main,
```
Comment (pacrob) [inline README.md:57]: The template and web3 are the only libs that have switched from `master` to `main` so far. The plan is to make the change at the same time as dropping py37, leading up to v7 web3 release.

**Comment by pacrob** [comment]
> I noticed we dont have linting configured for Python 3.7 but still build the wheel. Is that intentionally skipped?

Yes, py37 linting was borked and deemed not worth figuring out why, since all py37 testing will be dropped sooner than later.

> Also see the versions for flake8, etc are outdated but that could be another issue.

You can try bumping the versions and see what needs updating. If it's just a few files, I'd say go ahead and add it here. If it makes major changes, a new PR would be preferred.

**Comment by pacrob** [comment]
Looks great! Just moving `main` back to `master` in documentation. I also found a reference to just `doc` in `docs/overview.rst`. You could delete the `Join that chat at` + gitter link there too - gitter is long dropped in favor of discord.

**Review by pacrob — APPROVED** [review]
looks good, just the couple minor requested changes

**Review by pacrob — APPROVED** [review]
lgtm!

File: Makefile [hunk 2]
```diff
@@ -69,8 +69,8 @@ notes: check-bump
 	git commit -m "Compile release notes"
 
 release: check-bump clean
-	# require that you be on a branch that's linked to upstream/master
-	git status -s -b | head -1 | grep "\.\.upstream/master"
+	# require that upstream is configured for ethereum/<REPO_NAME>
+	git remote -v | grep "upstream\tgit@github.com:ethereum/<REPO_NAME>.git (push)\|upstream\thttps://github.com/ethereum/<REPO_NAME> (push)"
```
Comment (fselmo) [inline Makefile:73]: ❤️

**Review by fselmo — APPROVED** [review]
lgtm 👍🏼

**Pull request merged**